watchtower-plugin/src/convert.rs

```diff
@@ -258,6 +258,117 @@ impl TryFrom<serde_json::Value> for GetAppointmentParams {
     }
 }
 
+// Errors related to `getregistrationreceipt` command
+#[derive(Debug)]
+pub enum GetRegistrationReceiptError {
+    InvalidId(String),
+    InvalidFormat(String),
+}
+
+impl std::fmt::Display for GetRegistrationReceiptError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            GetRegistrationReceiptError::InvalidId(x) => write!(f, "{x}"),
+            GetRegistrationReceiptError::InvalidFormat(x) => write!(f, "{x}"),
+        }
+    }
+}
```

Comment:
There is no distinction in usage between these two different error variants. I think we can combine them into one.
We might also be able to get rid of them and use an `Err` instead.

Reply:
Actually I looked how other methods had this implemented and followed them 😅. Which way do you think would be better, keeping as it is, merging into one error or removing this entirely ? I will make changes accordingly.

Reply:
Yeah I noticed other param parsers are doing the same. I think we can make all of them to use `Err(String)||Err(&'static str)`.

> Which way do you think would be better, keeping as it is, merging into one error or removing this entirely ? I will make changes accordingly.

Try to remove it entirely, then we can do the same for other param errors in a follow up if it made sense.

Reply:
`Err` seems to be working fine, just had to tweak the code where parsing was done. I think I would wait for decision on [issue of Vec or Option<vec](https://github.com/talaia-labs/rust-teos/pull/199#discussion_r1181203980)  before final submission, rest changes are done.

Reply:
The reason why this is here is to make it properly testable. If you look at the other params, there are test suits to make sure the proper error is returned.

I guess we should add them too for `GetRegistrationReceipt` now

Reply:
> 

oh, got it. I will add tests for `GetRegistrationReceipt` params there. And in case of https://github.com/talaia-labs/rust-teos/pull/199#discussion_r1181203980, I think it would be better as Some(Vec<>) just because its common way in rust-teos code.

Reply:
> And in case of https://github.com/talaia-labs/rust-teos/pull/199#discussion_r1181203980, I think it would be better as Some(Vec<>) just because its common way in rust-teos code.

This would only make sense if https://github.com/talaia-labs/rust-teos/pull/199#discussion_r1188792055 is carried out. I am fine with both ways, but the `None` variant has to carry a meaning and not just be there so the interface look similar to other methods.

Reply:
Agreed